### ISSUE TITLE
The released API bundle is missing shared-libs

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#5068]
We use npm pack to zip up api and sentinel and all their bundledDependencies so when they get deployed on the server they just work. Currently when deploying the `medic:medic:master` release, API won't start due to a missing dependency on `@shared-libs/server-checks`. The [`medic:medic:master`](https://staging.dev.medicmobile.org/_couch/_utils/document.html?builds/medic%3Amedic%3Amaster) release on the market is missing this node_module, but it exists in the [`medic:medic:3.3.x`](https://staging.dev.medicmobile.org/_couch/_utils/document.html?builds/medic%3Amedic%3A3.3.x) release.

I suspect it has something to do with replacing [yarn with npm](https://github.com/medic/medic-webapp/issues/4814) which was done recently so start there.

**Comment by garethbowen** [comment]
Maybe caused by [this npm issue](https://github.com/npm/npm/issues/17708).

**Comment by SCdF** [comment]
I guess we were on an older version of npm before?

Suppose it's time to actually start publishing and versioning these

**Comment by garethbowen** [comment]
@SCdF Yes I think it's due to bumping npm to the latest. I think they would consider this a feature as it improves the integrity and reproducibility of the build because it doesn't include random stuff off your disk. So I don't think they're going to fix it in the short term.

That leaves us with a few possible solutions...

1. Publish our shared libs to npm and install them as normal npm dependencies. This is annoying for development if you're making a change in a shared-lib as well as somewhere else in the code as you'll need to publish the shared-lib first. We may be able to find ways to make this easier, such as overwriting the shared-libs with symlinks in development.
2. Do a custom step before calling `npm pack` to replace the file dependencies with inlined code somehow. This feels brittle and too bespoke so easy to forget about or break by accident.
3. Write a custom `npm pack` replacement. I had a quick look at this using the tar `--dereference` flag which inlines symlinks and it worked except because it doesn't understand anything about npm packaging we end up with lots of duplicate dependencies which blows the bundle out from ~5MB to ~30MB. I think it'd be complicated to do this properly.

After weighing these options up I think we should go with option 1 which more closely conforms with how I think npm intends this to be done which means it's more likely to be supported long term.

**Comment by SCdF** [comment]
Agree with 1 being the right option.

I think we can use `npm link` reasonably effectively. This allows easy development, and an example of it would be:
```sh
# First, setup the package to be linked
cd ./shared-libs/bulk-doc-utils
npm link
# Then, link it in API
cd ../../api
npm link @shared-libs/bulk-doc-utils
```
It may be worth scripting this so people don't have to understand it.

I'm less sure what to do about travis. On the one hand, we could have it *also* use link, but then we couldn't rework travis to use real builds, because `npm pack` doesn't get fixed by `npm link` (if only!). On the other hand, we should probably be using travis to publish shared libs (and horti, and medic-conf) anyway, because `npm publish` goes off what's on disk not what's in git, so doing it manually is asking for trouble, but I'm not sure how you'd work that in.

**Comment by garethbowen** [comment]
@ngaruko Make sure you can install the app.
